.github/workflows/python.yml

```diff
@@ -59,13 +60,35 @@ jobs:
       with:
         python-version: ${{ matrix.python-version }}
 
+    - name: Pre-build dependencies specific to Ubuntu
+      if: runner.os == 'Linux'
+      run: sudo apt install libsndfile1-dev
+
     - name: Pre-build dependencies
-      run: |
-        sudo apt install libsndfile1-dev
-        python -m pip install --upgrade pip
+      run: python -m pip install --upgrade pip
 
     - name: Install dependencies
       run: pip install -r requirements.txt
 
     - name: Test
       run: python test_porcupine.py ${{secrets.PV_VALID_ACCESS_KEY}}
+
+
+  build-self-hosted:
+    runs-on: ${{ matrix.machine }}
+
+    strategy:
+      matrix:
+        machine: [rpi2, rpi3, rpi4, jetson, beaglebone]
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Pre-build dependencies
+      run: python3 -m pip install --upgrade pip
+
+    - name: Install dependencies
+      run: pip3 install -r requirements.txt
+
+    - name: Test
+      run: python3 test_porcupine.py ${{secrets.PV_VALID_ACCESS_KEY}}
```

Comment:
Is there any reason we can't combine the two jobs seeing as how the steps are nearly incidental? I think it'll be easier to maintain.

Reply:
I reverted the extra commits.